src/stim/search/sat/wcnf.test.cc

```diff
@@ -0,0 +1,55 @@
+#include "stim/search/sat/wcnf.h"
+#include "gtest/gtest.h"
+
+using namespace stim;
+
+TEST(shortest_undetectable_logical_error_wcnf, no_error) {
+    // No error.
+    ASSERT_THROW(
+        { stim::shortest_undetectable_logical_error_wcnf(DetectorErrorModel()); },
+        std::invalid_argument);
+}
+
+TEST(shortest_undetectable_logical_error_wcnf, single_detector_single_observable) {
+  std::string wcnf = stim::shortest_undetectable_logical_error_wcnf(DetectorErrorModel(R"DEM(
+      error(0.1) D0 L0
+      error(0.1) D0
+    )DEM"));
+  // There should be 3 variables: x = (x_0, x_1, x_2)
+  // x_0 -- error 0 occurred
+  // x_1 -- error 1 occurred
+  // x_2 -- XOR of x_0 and x_1
+  // There should be 2 soft clauses:
+  // soft clause NOT(x_0) with weight 1
+  // soft clause NOT(x_0) with weight 1
+  // There should be 4 hard clauses to forbid the strings such that x_2 != XOR(x_0, x_1):
+  // hard clause x != (0, 0, 1)
+  // hard clause x != (0, 1, 0)
+  // hard clause x != (1, 0, 0)
+  // hard clause x != (1, 1, 1)
+  // Plus 1 hard clause to ensure detector is not flipped
+  // hard clause -x_2
+  // Plus 1 hard clause to ensure an observable is flipped:
+  // hard clause x_0
+  // This gives a total of 8 clauses
+  // The top value should be at least 1 + 1 + 1 = 3. In our implementation ends up being 9.
+  std::stringstream expected;
```

Comment:
Use a multiline string literal

```
expected = R"DIMACS(
so much space for activities!
)DIMACS";
```

Reply:
I don't understand why tabs in the data would prevent the use of multiline string literals.

Note that, for multiline literals, you don't indent the contents of the literal.

Reply:
I'm going to push back against this suggestion again because it prevents adding explanatory inline comments between each of the lines.